source/_drafts/Video-js-5-s-fluid-mode-and-playlist-picker.md

```diff
@@ -0,0 +1,208 @@
+---
+title: Video.js 5's fluid mode and playlist picker
+tags:
+  - videojs 5
+  - fluid
+---
+
+## How it works
+In videojs 5.0, we added support for truely fluid layouts with video.js.
+You can see an example of it on the [videojs website][website].
+
+It is done by using [intrinsic ratios][ratios]. In short, you make the video element take up 100% `width` and make the `height` be `0`.  We can set the player `width` to a particular size, say 50%; this way, as we resize the the browser window, the player should stay 50% the same of the window but keep it's aspect ratio for the video. The main trick is that in addition to setting the `width` on the player, we also apply a padding. Specifically, we want to set the following:
+```css
+.video-js {
+  padding-top: 56.25%
+}
+```
+This percentage is the aspect ratio of height with respect to width in a 16:9 video, the standard aspect ratio for most content now days.  What makes this work is that padding is set with respect to the width of the element, so the height will always be 56.25% of the width maintaining the aspec ratio that we want.
+
+## How to use it in video.js
+In video.js, to make a player fluid, you can either set the `fluid` option
+```js
+let player = videojs('preview-player', {
+  fluid: true
+});
+```
+Or you can add one of the fluid classes to the player: `.vjs-fluid`, `.vjs-4-3`, `.vjs-16-9`:
+```html
+<video id="preview-player" class="video-js vjs-fluid" controls data-setup={}>
+```
+`.vjs-4-3` maintains a 4:3 aspect ratio for the video and `.vjs-16-9` maintains a 16:9 one.  `.vjs-fluid` is a bit more special. It waits for the video metadata to load and then uses the video width and video height to calculate the correct aspect ratio to use for the video.
+
+## Playlist picker
+This works great if you only have the player by itself. What if you are trying to a attach a playlist to the video element and keep it at the same height? Like we did on the [advanced example page][advanced-example] on the videojs website. Like so:
```

Comment:
> Like we did on the [advanced example page][advanced-example] on the videojs website. 

Sentence fragment.

> Like so:

Getting a bit "like" heavy in these last two sentences :)